src/azure-cli/azure/cli/command_modules/synapse/operations/accesscontrol.py

```diff
@@ -119,6 +127,11 @@ def _resolve_object_id(cmd, assignee, fallback_to_object_id=False):
                            "If the assignee is a principal id, make sure the corresponding principal is created "
                            "with 'az ad sp create --id {assignee}'.".format(assignee=assignee))
 
+        if len(result) > 1:
```

Comment:
Add breaking change description at History Notes:

[Synapse] BREAKING CHANGE: az synapse role assignment create: When --assignee can't uniquely determine the principal object, the command will raise error instead of adding a role assignment for the uncertain principal object.